spec/acceptance/mssql_spec.rb

```diff
@@ -0,0 +1,374 @@
+require 'acceptance_spec_helper'
+
+RSpec.describe 'MSSQL sessions and MSSQL modules' do
+  include_context 'wait_for_expect'
+
+  TESTS = {
+    mssql: {
+      target: {
+        session_module: "auxiliary/scanner/mssql/mssql_login",
+        type: 'MSSQL',
+        platforms: [:linux, :osx, :windows],
+        datastore: {
+          global: {},
+          module: {
+            username: ENV.fetch('MSSQL_USER', 'sa'),
+            password: ENV.fetch('MSSQL_PASSWORD', 'yourStrong(!)Password'),
+            rhost: ENV.fetch('MSSQL_RHOST', '127.0.0.1'),
+            rport: ENV.fetch('MSSQL_RPORT', '1433'),
+            database: 'master'
+          }
+        }
+      },
+      module_tests: [
+        {
+          name: "post/test/mssql",
+          platforms: [:linux, :osx, :windows],
+          targets: [:session],
+          skipped: false,
+        },
+        {
+          name: "auxiliary/scanner/mssql/mssql_hashdump",
+          platforms: [:linux, :osx, :windows],
+          targets: [:session, :rhost],
+          skipped: false,
+          lines: {
+            all: {
+              required: [
+                'Instance Name:',
+                'Scanned 1 of 1 hosts (100% complete)'
+              ]
+            },
+          }
+        },
+        {
+          name: "auxiliary/admin/mssql/mssql_enum",
+          platforms: [:linux, :osx, :windows],
+          targets: [:session, :rhost],
+          skipped: false,
+          lines: {
+            all: {
+              required: [
+                'Version:',
+                /Microsoft SQL Server \d+.\d+/,
+                'Databases on the server:',
+                'System Logins on this Server:'
+              ]
+            },
+          }
+        },
+        {
+          name: "auxiliary/scanner/mssql/mssql_schemadump",
+          platforms: [:linux, :osx, :windows],
+          targets: [:session, :rhost],
+          skipped: false,
+          lines: {
+            all: {
+              required: [
+                'Instance Name:',
+                'Scanned 1 of 1 hosts (100% complete)'
```

Comment:
When a module fails silently in the background, this message will still be printed so this particular line does not really add anything.